src/main/java/rx/observables/BlockingObservable.java

```diff
@@ -477,4 +473,227 @@ private void awaitForComplete(CountDownLatch latch, Subscription subscription) {
+     * <p>
+     * The unsubscription and backpressure is composed through.
+     * @param subscriber the subscriber to forward events and calls to in the current thread
+     */
+    @Experimental
+    public void subscribe(Subscriber<? super T> subscriber) {
+        final NotificationLite<T> nl = NotificationLite.instance();
+        final BlockingQueue<Object> queue = new LinkedBlockingQueue<Object>();
+        final Producer[] theProducer = { null }; 
+        
+        Subscriber<T> s = new Subscriber<T>() {
+            @Override
+            public void onNext(T t) {
+                queue.offer(nl.next(t));
+            }
+            @Override
+            public void onError(Throwable e) {
+                queue.offer(nl.error(e));
+            }
+            @Override
+            public void onCompleted() {
+                queue.offer(nl.completed());
+            }
+            
+            @Override
+            public void setProducer(Producer p) {
+                theProducer[0] = p;
+                queue.offer(SET_PRODUCER);
+            }
+            
+            @Override
+            public void onStart() {
+                queue.offer(ON_START);
+            }
+        };
+        
+        subscriber.add(s);
+        subscriber.add(Subscriptions.create(new Action0() {
+            @Override
+            public void call() {
+                queue.offer(UNSUBSCRIBE);
+            }
+        }));
+        
+        o.subscribe(s);
+        
+        try {
+            for (;;) {
+                if (subscriber.isUnsubscribed()) {
+                    break;
+                }
+                Object o = queue.poll();
+                if (o == null) {
+                    o = queue.take();
+                }
+                if (subscriber.isUnsubscribed() || o == UNSUBSCRIBE) {
+                    break;
+                }
+                if (o == ON_START) {
+                    subscriber.onStart();
+                } else
+                if (o == SET_PRODUCER) {
+                    subscriber.setProducer(theProducer[0]);
+                } else
+                if (nl.accept(subscriber, o)) {
+                    return;
+                }
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            subscriber.onError(e);
+        } finally {
+            s.unsubscribe();
+        }
+    }
+    
+    /**
+     * Runs the source observable to a terminal event, ignoring any values and rethrowing any exception.
+     */
+    @Experimental
+    public void subscribe() {
+        run();
+    }
+    
+    /**
+     * Subscribes to the source and calls the given action on the current thread and rethrows any exception wrapped
+     * into OnErrorNotImplementedException.
+     * @param onNext the callback action for each source value
+     */
+    @Experimental
+    public void subscribe(final Action1<? super T> onNext) {
```

Comment:
They have different behavior. `forEach` executes the action on the thread that emits the `onNext` events and awaits the terminal event in the current thread. This method calls the action on the current thread.